### PR TITLE
fix: fixed formatting when a field is undefined

### PR DESCRIPTION
**Pull request by marco-ippolito**
resolves: https://github.com/nearform/github-board-slack-notifications/issues/124
the aim is to prevent undefined fields from appearing in the message

### CODE REVIEW

File: config/notification-config.json [hunk 1]
```diff
@@ -1,27 +1,27 @@
 {
     "actions": {
         "created": {
-            "message": "💡 <{authorUrl}| {authorName}> has a created a {content_type} titled _<{itemUrl}| #{itemNumber} {title}>_ in <{projectUrl}| {projectName}> ➕️"
+            "message": "💡 <{authorUrl}{authorName}> has a created a {content_type} titled _<{itemUrl}{itemNumber}{title}>_ in <{projectUrl}{projectName}> ➕️"
```
Comment (simoneb) [inline config/notification-config.json:4]: "has a created" -> "has created"

**Review by simoneb — CHANGES_REQUESTED** [review]
let's move all the icons at the beginning of the sentences, instead of the end

File: config/notification-config.json [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
-            "message": "💡 {content_type} _<{itemUrl}| #{itemNumber} {title}>_ has been deleted from <{projectUrl}| {projectName}> ❌"
+            "message": "💡 {content_type} _<{itemUrl}{itemNumber}{title}>_ has been deleted from <{projectUrl}{projectName}> ❌"
         },
         "converted": {
-            "message": "💡 {content_type} _<{itemUrl}| #{itemNumber} {title}>_ has been converted 🔄",
+            "message": "💡 {content_type} _<{itemUrl}{itemNumber}{title}>_ has been converted 🔄",
             "changes": {
                 "Issue": {
-                    "message": "💡 <{authorUrl}| {authorName}> has a created a {content_type} titled _<{itemUrl}| #{itemNumber} {title})}>_ in <{projectUrl}| {projectName}> 🚩"
+                    "message": "💡 <{authorUrl}{authorName}> has a created a {content_type} titled _<{itemUrl}{itemNumber}{title}>_ in <{projectUrl}{projectName}> 🚩"
```
Comment (simoneb) [inline config/notification-config.json:13]: has a created -> has created

File: src/messages.js [hunk 3]
```diff
@@ -52,19 +52,29 @@ export function formatMessage({ content_type, node, message, extra }) {
 function getKeysFromNode({ node }) {
   const { creator, project, content, fieldValueByName } = node
   return {
-    authorUrl: content?.author?.url ? content.author.url : creator?.url,
+    authorUrl: addPipeToUrl(
+      content?.author?.url ? content.author.url : creator?.url
```
Comment (JohanX) [inline src/messages.js:56]: Using both optional chaining and ternary operators seems a bit much. Maybe do optional chaining with `||`

Comment (marco-ippolito) [reply src/messages.js:56]: there is a case where content is undefined so it might cause some trouble

Comment (JohanX) [reply src/messages.js:56]: I meant `authorUrl: addPipeToUrl(content?.author?.url || creator?.url)`

Comment (marco-ippolito) [reply src/messages.js:56]: yes that's better I'll do the same for authorName

**Review by simoneb — APPROVED** [review]

**Pull request merged**